public/app/plugins/datasource/cloudwatch/datasource.js

```diff
@@ -416,7 +416,16 @@ function (angular, _, moment, dateMath, kbn, templatingVariable, CloudWatchAnnot
       return _.chain(targets)
       .map(function(target) {
         var dimensionKey = _.findKey(target.dimensions, function(v) {
-          return templateSrv.variableExists(v) && !_.has(scopedVars, templateSrv.getVariableName(v));
+          if (!templateSrv.variableExists(v)) {
+            return false;
+          }
+
+          templateSrv.getVariableNames(v).forEach(function(variableName) {
+            if (_.has(scopedVars, variableName)) {
+              return false;
+            }
+          });
+          return true;
```

Comment:
I am sorry I still do not understand how this is supposed to work. Was a mistake to let cloudwatch use templating like it was never designed for. 

Can you describe in words what this functions does? and each line?
To me it looks like this

var dimensionKey = _.findKey(target.dimensions, function(v) {
// loop over all dimensions, find the first one that contains a variable, but not if the variable exist in local scope. 
// why only find the first dimension that matches criteria? 

// 

```
templateSrv.getVariableNames(v).forEach(function(variableName) {
            if (_.has(scopedVars, variableName)) {
              return false;
            }
       });
```
this code looks like it does nothing? loops over variables and then return true or false, but since the return is inside a function this is actually doing nothing.